avc/src/color_detector_avc.cpp

```diff
@@ -15,184 +15,86 @@ using namespace ros;
+    inRange(frameHSV, low, high, obstacleImg);
+    
+    //ROS_INFO_STREAM(obstacleImg.rows << ", " << obstacleImg.cols << ", " << mask.rows << ", " << mask.cols);
+    Mat masked;
+    obstacleImg.copyTo(masked, mask);
 
-    auto kernel_size = 5;
-    Mat erosion_kernel = getStructuringElement(MORPH_CROSS, Size(kernel_size, kernel_size));
-
-    erode(ramp, ramp, erosion_kernel);
-
-    return ramp;
+    return masked;
 }
 
 void ImageRectCB(const sensor_msgs::ImageConstPtr& msg) {
 
     cv_bridge::CvImagePtr cv_ptr;
     Mat frame;
-    Mat output;
 
     try {
         cv_ptr = cv_bridge::toCvCopy(msg, "bgr8");
     } catch (cv_bridge::Exception& e) {
         ROS_ERROR("CV-Bridge error: %s", e.what());
         return;
     }
+	frame = cv_ptr->image;
+    Mat output(frame.rows, frame.cols, CV_8UC1, Scalar::all(0));
 
-    //applying detectCurb() function to image
-    frame = cv_ptr->image;
-    Mat curbs = detectCurb(frame);
-    Mat hoop = detectHoop(frame);
-    Mat ramp = detectRamp(frame);
-    output = curbs + hoop + ramp;
+	for (int i = 0; i < lows.size(); i++) {
+		Mat partial;
+		partial = detectObstacleColor(frame, lows[i], highs[i]);
+		bitwise_or(output, partial, output);
+	}
 
     sensor_msgs::Image outmsg;
     cv_ptr->image = output;
-    cv_ptr->encoding = "bgr8";
+    cv_ptr->encoding = "mono8";
     cv_ptr->toImageMsg(outmsg);
     img_pub.publish(outmsg);
 }
 
-void ColorsTransformedCB(const sensor_msgs::ImageConstPtr& msg) {
-    float pxPerMeter = 100.0;
-    cv_bridge::CvImagePtr cv_ptr;
-    try {
-        cv_ptr = cv_bridge::toCvCopy(msg, "bgr8");
-    } catch(cv_bridge::Exception& e) {
-        ROS_ERROR("CV_Bridge error: %s", e.what());
-        return;
-    }
-
-    pcl::PointCloud<pcl::PointXYZ>::Ptr cloud(new pcl::PointCloud<pcl::PointXYZ>);
-    Mat transformed = cv_ptr->image;
-    cvtColor(transformed, transformed, CV_BGR2GRAY);
-
-    for(int r = 0; r < transformed.rows; r++) {
-        uchar* row = transformed.ptr<uchar>(r);
-        for(int c = 0; c < transformed.cols; c++) {
-            if(row[c]) {
-                pcl::PointXYZ point;
-                point.y = -1 * (c - transformed.cols / 2.0f) / pxPerMeter;
-                point.x = (transformed.rows - r) / pxPerMeter;
-                point.z = 0.0;
-
-                cloud->push_back(point);
-            }
-        }
-    }
-
-    pcl::PCLPointCloud2 cloud_pc2;
-    pcl::toPCLPointCloud2(*cloud, cloud_pc2);
-    sensor_msgs::PointCloud2 cloud_msg;
-    pcl_conversions::fromPCL(cloud_pc2, cloud_msg);
-    cloud_msg.header.frame_id = "base_footprint";
-    cloud_msg.header.stamp = ros::Time::now();
-    cloud_pub.publish(cloud_msg);
-}
 
 int main(int argc, char** argv) {
 
     init(argc, argv, "color_detector_avc");
 
     //Doesn't proccess the top half of the picture
     vector<Mat> mask_segments = {
-        Mat(1080,960,CV_8UC3, Scalar::all(1)),
-        Mat::zeros(1080,960,CV_8UC3)
+        Mat::zeros(45,160,CV_8UC1), 
```

Comment:
We might want to parameterize these since this is now intended to be more general purpose as a node